sql/src/main/java/io/crate/planner/node/ddl/CreateTablePlan.java

```diff
@@ -0,0 +1,243 @@
+
+    private static final String CLUSTERED_BY_IN_PARTITIONED_ERROR = "Cannot use CLUSTERED BY column in PARTITIONED BY clause";
+
+    private final AnalyzedCreateTable createTable;
+    private final NumberOfShards numberOfShards;
+    private final TableCreator tableCreator;
+    private final Schemas schemas;
+
+    public CreateTablePlan(AnalyzedCreateTable createTable,
+                           NumberOfShards numberOfShards,
+                           TableCreator tableCreator,
+                           Schemas schemas) {
+        this.createTable = createTable;
+        this.numberOfShards = numberOfShards;
+        this.tableCreator = tableCreator;
+        this.schemas = schemas;
+    }
+
+    @Override
+    public StatementType type() {
+        return StatementType.DDL;
+    }
+
+    @Override
+    public void executeOrFail(DependencyCarrier dependencies,
+                              PlannerContext plannerContext,
+                              RowConsumer consumer,
+                              Row params,
+                              SubQueryResults subQueryResults) {
+        CreateTableAnalyzedStatement stmt = createStatement(
+            createTable,
+            plannerContext.transactionContext(),
+            plannerContext.functions(),
+            params,
+            subQueryResults,
+            numberOfShards,
+            schemas,
+            dependencies.fulltextAnalyzerResolver());
+
+        if (stmt.noOp()) {
+            consumer.accept(InMemoryBatchIterator.empty(SENTINEL), null);
+            return;
+        }
+
+        tableCreator.create(stmt)
+            .whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
+    }
+
+    @VisibleForTesting
+    public static CreateTableAnalyzedStatement createStatement(AnalyzedCreateTable createTable,
+                                                               CoordinatorTxnCtx txnCtx,
+                                                               Functions functions,
+                                                               Row params,
+                                                               SubQueryResults subQueryResults,
+                                                               NumberOfShards numberOfShards,
+                                                               Schemas schemas,
+                                                               FulltextAnalyzerResolver fulltextAnalyzerResolver) {
+        Function<? super Symbol, Object> eval = x -> SymbolEvaluator.evaluate(
+            txnCtx,
+            functions,
+            x,
+            params,
+            subQueryResults
+        );
+        CreateTable<Symbol> table = createTable.createTable();
+        RelationName relationName = createTable.relationName();
+        GenericProperties<Object> properties = table.properties().map(eval);
+
+        CreateTableAnalyzedStatement stmt = new CreateTableAnalyzedStatement();
```

Comment:
Do you intend to adapt theses classes further later on or is the plan to keep this as is ?

Reply:
ok, I think we can also follow up on this later on.